R/title.R

```diff
@@ -0,0 +1,95 @@
+#' Generate Application Name
+#'
+#' Generate a short name (identifier) for an application given an application
+#' title.
+#'
+#' @param appTitle A descriptive title for the application.
+#' @param appPath The path to the application's content, either a directory
+#'   or an individual document. Optional.
+#' @param account The account where the application will be deployed. Optional.
+#' @return
+#' Returns a valid short name for the application.
+#'
+#' @details
+#' This function modifies the title until it forms a suitable application name.
+#' Suitable application names are 3 - 64 characters long and contain only
+#' alphanumeric characters.
+#'
+#' The function is intended to be used to find a name for a new application.
+#' If \code{appPath} and \code{account} are both specified, then the returned
+#' name will also be unique among locally known deployments of the directory
+#' (note that it is not guaranteed to be unique on the server).
+#'
+#' @examples
+#' # Generate a short name for a sample application
+#' generateAppName("My Father's Country", "~/fathers-country", "myacct")
+#'
+#' @export
+
+generateAppName <- function(appTitle, appPath = NULL, account = NULL) {
+  munge <- function (title) {
+    # safe default if no title specified
+    if (is.null(title)) {
+      return("")
+    }
+
+    # start by removing most non-Latin characters and converting to lowercase
+    name <- tolower(gsub("[^A-Za-z0-9\\-_ ]+", "", title))
+
+    # replace spaces with underscores
+    name <- gsub(" ", "_", name, fixed = TRUE)
+
+    # trim to 64 characters
```

Comment:
It would be interesting to consider trimming to the nearest word instead of simply cutting the string at 64 characters. For example, `...uses_of_the_oxford_comma`, instead of being truncated at `..uses_of_the_ox`, would become `...uses_of_the`. This could help eliminate misleading (or obscene) app names, but possibly at the expense of clarity in some cases.